geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreTest.scala

```diff
@@ -235,6 +237,24 @@ class GeoMesaDataStoreTest extends Specification {
       ds.stats.getBounds(sft) mustEqual
           new ReferencedEnvelope(feature.getDefaultGeometry.asInstanceOf[Point].getEnvelopeInternal, CRS_EPSG_4326)
     }
+    "prioritize temporal filter plans" in {
+      val spec = "name:String:index=true,age:Int,dtg:Date,*geom:Point:srid=4326"
+
+      val ds = new TestGeoMesaDataStore(true)
+      ds.createSchema(SimpleFeatureTypes.createType("default", spec))
+      ds.createSchema(SimpleFeatureTypes.createType("temporal", s"$spec;geomesa.temporal.priority=true"))
+
+      val filters = Seq(
+        ("name like 'a%' and dtg DURING 2020-01-01T00:00:00.00Z/2020-01-02T00:00:00.00Z", AttributeIndex, Z3Index),
+        ("name > 'a' AND name < 'b' and dtg DURING 2020-01-01T00:00:00.00Z/2020-01-02T00:00:00.00Z", AttributeIndex, Z3Index),
+        ("name = 'a' and dtg DURING 2020-01-01T00:00:00.00Z/2020-01-02T00:00:00.00Z", AttributeIndex, AttributeIndex)
```

Comment:
Would it be 'easy' to add tests for the id cases above?

Reply:
easy enough - but id filters are always #1 priority so it wouldn't show any change. i guess that would be a meaningful result in itself. i'll add one

Reply:
added